server/data_common/dataset_metadata.py

```diff
@@ -27,47 +27,47 @@ def request_dataset_metadata_from_data_portal(data_portal_api_base: str, explore
         return None
 
 
-def extrapolate_dataset_location_from_config(server_config: ServerConfig, dataset_explorer_location: str):
+def infer_dataset_s3_uri(server_config: ServerConfig, dataset_root: str, dataset_id: str):
     """
-    Use the dataset_explorer_location and the server config to determine where the dataset is stored
+    Use the dataset_root, dataset_id, and the server config to infer the physical S3 URI for an Explorer dataset
+    artifact
     """
     # TODO @mdunitz remove after fork, update config to remove single_dataset option, the multiroot lookup will need to
     #  remain while we support covid 19 cell atlas.
     #   See ticket https://app.zenhub.com/workspaces/single-cell-5e2a191dad828d52cc78b028/issues/chanzuckerberg/corpora-data-portal/1281 # noqa
     if server_config.single_dataset__datapath:
-        datapath = server_config.single_dataset__datapath
-        return datapath
-    else:
-        dataset_explorer_location = dataset_explorer_location.split("/")
-        dataset = dataset_explorer_location.pop(-1)
-        url_dataroot = "/".join(dataset_explorer_location)
-        dataroot = None
-        for dataroot_dict in server_config.multi_dataset__dataroot.values():
-            if dataroot_dict["base_url"] == url_dataroot:
-                dataroot = dataroot_dict["dataroot"]
-                break
-        if dataroot is None:
-            raise DatasetAccessError(f"Invalid dataset {url_dataroot}/{dataset}")
-        datapath = path_join(dataroot, dataset)
-        # path_join returns a normalized path.  Therefore it is
-        # sufficient to check that the datapath starts with the
-        # dataroot to determine that the datapath is under the dataroot.
-        if not datapath.startswith(dataroot):
-            raise DatasetAccessError(f"Invalid dataset {url_dataroot}/{dataset}")
-        return datapath
-
-
-def get_dataset_metadata_for_explorer_location(dataset_explorer_location: str, app_config: AppConfig):
+        return server_config.single_dataset__datapath
+
+    for dataroot_dict in server_config.multi_dataset__dataroot.values():
+        if dataroot_dict["base_url"] == dataset_root:
+            dataroot = dataroot_dict["dataroot"]
+            break
+    if dataroot is None:
+        raise DatasetAccessError(f"Invalid dataset root {dataset_root}")
+
+    s3_uri = path_join(dataroot, dataset_id)
+
+    # TODO: Is the check still necessary?
+    # path_join returns a normalized path.  Therefore it is
+    # sufficient to check that the datapath starts with the
+    # dataroot to determine that the datapath is under the dataroot.
+    if not s3_uri.startswith(dataroot):
+        raise DatasetAccessError(f"Invalid dataset {s3_uri}")
+    return s3_uri
```

Comment:
Refactored and simplified `extrapolate_dataset_location_from_config`, renaming to `infer_dataset_s3_uri`. In particular, split the `dataset_explorer_location` param into separate `dataset_root` and `dataset_id` params. This is intended to make it clearer what specifically the method needs, and these are common params throughout the code base. The `dataset_explorer_location` param was not descriptive enough, and callers were having to construct it from the separate params.